### PR TITLE
[FLINK-25061][tests] Add assertj in flink-parent and link the code quality guide in the PR template 

### PR DESCRIPTION
**Pull request by slinkydeveloper**
For more details about this PR, check out: https://lists.apache.org/thread/33t7hz8w873p1bc5msppk65792z08rgt

Changes:

* Add assertj in the dependency management
* Add our code quality guide link in the "verifying this change" section of the PR template

### CODE REVIEW

**Review by matriv — APPROVED** [review]
LGTM, thx!

File: pom.xml [hunk 1]
```diff
@@ -586,6 +587,13 @@ under the License.
 				<version>${junit4.version}</version>
 			</dependency>
 
+			<dependency>
+				<groupId>org.assertj</groupId>
+				<artifactId>assertj-core</artifactId>
+				<!-- use 2.9.1 for Java 7 projects -->
```
Comment (zentol) [inline pom.xml:593]: this line doesn't seem useful, surely we aren't going to revert to Java 7.

Comment (slinkydeveloper) [reply pom.xml]: You never know :rofl:

**Comment by zentol** [comment]
@flinkbot run azure

File: pom.xml [hunk 2]
```diff
@@ -586,6 +587,12 @@ under the License.
 				<version>${junit4.version}</version>
 			</dependency>
 
+			<dependency>
+				<groupId>org.assertj</groupId>
+				<artifactId>assertj-core</artifactId>
```
Comment (zentol) [inline pom.xml:592]: If the goal is to use assertj everywhere, then we should add it as a general test dependency as we did for junit/hamcrest.

**Review by zentol — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (zentol) [reply pom.xml:592]: Since this also changes the dependency tree, I think we should add a ticket as well.

**Comment by zentol** [comment]
@flinkbot run azure

**Pull request merged**